docs/concepts/feathr-concepts-for-beginners.md

```diff
@@ -0,0 +1,115 @@
+
+Since this is a process with "join", you need to specify the observation data source and the features you want to get, and the result will be the observation data plus the additional features.
+
+```python
+feature_query = FeatureQuery(
+    feature_list=["f_location_avg_fare", "f_trip_time_rounded", "f_is_long_trip_distance"], key=location_id)
+settings = ObservationSettings(
+    observation_path="wasbs://public@azurefeathrstorage.blob.core.windows.net/sample_data/green_tripdata_2020-04.csv",
+    event_timestamp_column="lpep_dropoff_datetime",
+    timestamp_format="yyyy-MM-dd HH:mm:ss")
+client.get_offline_features(observation_settings=settings,
+                            feature_query=feature_query,
+                            output_path=output_path)
+```
+
+## What is "materialization" in Feathr?
+
+You are very likely to train a machine learning model with the features that you just queried (with `get_offline_features()`). After you have trained a machine learning model, say a fraud detection model, you are likely to put the machine learning model into an online envrionment and do online inference.
+
+In that case, you will need to retrieve the features (for example the user historical spending) in real time, since the fraud detection model is very time sensitive. Usually some key-value store is used for that scenario (for example Redis), and Feathr will help you to materialize features in the online environment for faster inference. That is why you will see something like below, where you specify Redis as the online storage you want to use, and retrieve features from online envrionment from there:
+
+```python
+redisSink = RedisSink(table_name="nycTaxiDemoFeature")
+# Materialize two features into a redis table.
+settings = MaterializationSettings("nycTaxiMaterializationJob",
+sinks=[redisSink],
+feature_names=["f_location_avg_fare", "f_location_max_fare"])
+client.materialize_features(settings)
+client.get_online_features(feature_table = "agg_features",
+                           key = "265",
+                           feature_names = ['f_location_avg_fare', 'f_location_max_fare'])
+```
+
+## Illustration
+
+An illustration of the concepts that we are going to talk about is like this:
+![Feature Join Process](../images/observation_data.png)
+
+## Point in time joins and aggregations
+
+Assuming users are already familar with the "regular" joins, for example inner join or outer join, and in many of the use cases, we care about time, that is why in Feathr we provide a capability called Point in time Join (and with other time based aggregations). That is also a reason why we focus very much on time and need end users to specify the timestamp column.
```

Comment:
Explicitly states that the feature timestamp is smaller than observation timestamp and this avoids information leak, i.e. using future data to train model at some time in the past, etc.